.github/workflows/publish-docker-main.yml

```diff
@@ -52,17 +51,13 @@ jobs:
         run: make generate
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
-      - name: Build docker image
-        run: |
-          make docker || make docker
       - name: Log in to the Container registry
         uses: docker/login-action@v1.10.0
-        if: github.ref == 'refs/heads/main'
         with:
           registry: ${{ env.HUB }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Push docker image
-        if: github.ref == 'refs/heads/main'
+      - name: Build and Push docker image
         run: |
-          make docker.push || make docker.push
\ No newline at end of file
+          make docker || make docker
+          make docker.push || make docker.push
```

Comment:
The previous workflow failed to push the Docker images is not because `make docker` and `make docker.push` are in different jobs, it's because `make docker` doesn't even build the Docker images, which is not reproducible in my local machine.

See this https://github.com/apache/skywalking-banyandb/runs/3349244592#step:8:378 , there is no Docker images after `make docker`

Reply:
> The previous workflow failed to push the Docker images is not because `make docker` and `make docker.push` are in different jobs, it's because `make docker` doesn't even build the Docker images, which is not reproducible in my local machine.
> 
> See this https://github.com/apache/skywalking-banyandb/runs/3349244592#step:8:378 , there is no Docker images after `make docker`

There is a warning at the very beginning of the step. The docker image built only exists in the current layer. So we need to build and push in one step.

Reply:
https://github.com/apache/skywalking-banyandb/runs/3349154715#step:8:12

Reply:
@lujiajing1126 do you mean this `fatal: No names found, cannot describe anything.
30`? It's a git command to set a variable from what I understand and should not affect the docker task.

Reply:
```
time="2021-08-17T09:53:39Z" level=warning msg="No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load"
```

Reply:
@lujiajing1126 check this https://github.com/apache/skywalking-banyandb/runs/3349370271 , I made your changes in this PR run in [my branch](https://github.com/apache/skywalking-banyandb/tree/test), it doesn't help at all

Reply:
I've added `--load` options in Makefile.

Pls check the latest CI log. It seems fine now.